File: Zadania z PDF'ów/Szyfr Cezara/c++/Szyfrowanie uwzględnieniem polskich znaków diakrytycznych.cpp
#include <iostream>
#include <string>

using namespace std;

const string alfabet_m = "aąbcćdeęfghijklłmnńoópqrsśtuvwxyzźż";
const string alfabet_w = "AĄBCĆDEĘFGHIJKLŁMNŃOÓPQRSŚTUVWXYZŹŻ";

string Cezar_PL(string s, int klucz)
{
    string szyfrogram = "";
    char litera;
    int nowy_indeks;

    for (int i = 0; i < s.length(); i++)
    {
        litera = s[i];

        if (alfabet_m.find(litera) != string::npos)
        {
            nowy_indeks = (alfabet_m.find(litera) + klucz) % alfabet_m.length();
            szyfrogram += alfabet_m[nowy_indeks];
        }
        else if (alfabet_w.find(litera) != string::npos)
        {
            nowy_indeks = (alfabet_w.find(litera) + klucz) % alfabet_w.length();
            szyfrogram += alfabet_w[nowy_indeks];
        }
        else
        {
            szyfrogram += litera;
        }
    }

    return szyfrogram;
}

int main()
{
    cout << Cezar_PL("test", 3) << endl;
    return 0;
}

File: Zadania z PDF'ów/Szyfr Cezara/Python/Szyfrowanie uwzględnieniem polskich znaków diakrytycznych.py
alfabet_m = "aąbcćdeęfghijklłmnńoópqrsśtuvwxyzźż"
alfabet_w = "AĄBCĆDEĘFGHIJKLŁMNŃOÓPQRSŚTUVWXYZŹŻ"

klucz = 0
jawny = ""
szyfrogram = ""

jawny = str(input("Podaj tekst do zaszyfrowania: "))
klucz = int(input("Podaj klucz: "))

for i in range(0, len(jawny)):
    litera = jawny[i]

    if litera in alfabet_m:
        nowy_indeks = (alfabet_m.find(litera) + klucz) % len(alfabet_m)
        szyfrogram += alfabet_m[nowy_indeks]
    elif litera in alfabet_w:
        nowy_indeks = (alfabet_w.find(litera) + klucz) % len(alfabet_w)
        szyfrogram += alfabet_w[nowy_indeks]
    else:
        szyfrogram += litera  # Jeśli znak nie jest w alfabecie (np. spacje, cyfry)

print("Zaszyfrowany tekst:", szyfrogram)
